### PR TITLE
Use column_views instead of column_device_views in binary operations.

### PR DESCRIPTION
**Pull request by bdice**
This PR changes the internal APIs used for binary operations to use `column_view` objects instead of `column_device_view` objects. This change is needed for the eventual support of structs in binary operations. See also: PR #9452.

### CODE REVIEW

File: cpp/src/binaryop/compiled/binary_ops.hpp [hunk 1]
```diff
@@ -175,45 +175,45 @@ bool is_supported_operation(data_type out, data_type lhs, data_type rhs, binary_
...
+ * @param rhs view of right operand column
+ * @param is_lhs_scalar true if @p lhs is a single element column representing a scalar
+ * @param is_rhs_scalar true if @p rhs is a single element column representing a scalar
  * @param op comparison binary operator
  * @param stream CUDA stream used for device memory operations
  */
-void dispatch_equality_op(mutable_column_device_view& outd,
-                          column_device_view const& lhsd,
-                          column_device_view const& rhsd,
+void dispatch_equality_op(mutable_column_view& outd,
```
Comment (rwlee) [inline cpp/src/binaryop/compiled/binary_ops.hpp:214]:
```suggestion
void dispatch_equality_op(mutable_column_view& out,
```

File: cpp/src/binaryop/compiled/binary_ops.hpp [hunk 2]
```diff
@@ -175,45 +175,45 @@ bool is_supported_operation(data_type out, data_type lhs, data_type rhs, binary_
...
  * @p outd type is boolean.
  *
  * This template is instantiated for each binary operator.
  *
- * @param outd mutable device view of output column
- * @param lhsd device view of left operand column
- * @param rhsd device view of right operand column
- * @param is_lhs_scalar true if @p lhsd is a single element column representing a scalar
- * @param is_rhs_scalar true if @p rhsd is a single element column representing a scalar
+ * @param outd mutable view of output column
```
Comment (rwlee) [inline cpp/src/binaryop/compiled/binary_ops.hpp:206]:
```suggestion
 * @param out mutable view of output column
```

**Review by rwlee — APPROVED** [review]
Small nits, otherwise looks good and matches functionality implemented and already reviewed in #9452

**Comment by vyasr** [comment]
@bdice same request as  #10776, could you write a more descriptive PR description.

**Review by jrhemstad — APPROVED** [review]

**Comment by bdice** [comment]
@gpucibot merge

**Comment by bdice** [comment]
rerun tests

**Pull request merged**